xonsh/ptk/shell.py

```diff
@@ -77,7 +77,8 @@ def singleline(self, store_in_history=True, auto_suggest=None,
                     enable_history_search=enable_history_search,
                     reserve_space_for_menu=0,
                     key_bindings_registry=self.key_bindings_manager.registry,
-                    display_completions_in_columns=multicolumn)
+                    display_completions_in_columns=multicolumn,
+                    vi_mode=builtins.__xonsh_env__.get('VI_MODE'))
```

Comment:
Can you make this a little more robust, please, by testing the `xonsh.shell.prompt_toolkit_version()` to figure out which kwarg to use? That would be great!